packages/maestro-provider/src/api/api.ts

```diff
@@ -73,7 +88,20 @@ export const api = (
 		setImmediate(async () => {
 			try {
 				for await (const items of repository(repoInfo).getOrganizationRecords({ organization })) {
-					indexer.bulkUpsert(repoInfo.indexName, items);
+					if (items.length > 0) {
+						if (repoInfo.type === RepositoryType.SONG) {
+							// convert Song documents into fileCentric or analysisCentric document
+							const converted = convertAnalyses(repoInfo, items);
+
+							if (converted.length > 0) {
+								indexer.bulkUpsert(repoInfo.indexName, converted);
+							} else {
+								logger.error(`Error converting records into '${repoInfo.indexingMode}Centric document'`);
+							}
+						} else {
+							indexer.bulkUpsert(repoInfo.indexName, items);
+						}
+					}
```

Comment:
DRY: this code seems to mostly identical to what's in lines 45-58. looks like it could be abstracted into a function that takes a converter function, or something like that